### ISSUE TITLE
[icetray] memory tracking and "losing" track (Trac #2083)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2075]
In certain situations, where we allocate memory in c++ but delete in python, we lose track of the memory.

To solve this, we need a bigger hammer.  Specifically, we need to override `malloc`/`free` instead of `new`/`delete`.  The easiest way is to LD_PRELOAD the new functions in, such as how [https://github.com/KDE/heaptrack/tree/master/src/track heaptrack does it].  This seems to be reliable, and might get rid of one of the segfault issues.

Migrated from https://code.icecube.wisc.edu/ticket/2083
```json
{
    "status": "closed", 
    "changetime": "2019-02-13T14:14:51", 
    "description": "In certain situations, where we allocate memory in c++ but delete in python, we lose track of the memory.\n\nTo solve this, we need a bigger hammer.  Specifically, we need to override `malloc`/`free` instead of `new`/`delete`.  The easiest way is to LD_PRELOAD the new functions in, such as how [https://github.com/KDE/heaptrack/tree/master/src/track heaptrack does it].  This seems to be reliable, and might get rid of one of the segfault issues.", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1550067291068818", 
    "component": "combo core", 
    "summary": "[icetray] memory tracking and \"losing\" track", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2017-08-25T20:59:27", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20170913T18:55:21: 
* david.schultz commented:

> In changeset 2966:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2966"
> remove icetray memory code. this will be replaced by an external tool using LD_PRELOAD, with one helper function in icetray to change the scope for each module. closes #2075
> ```


* david.schultz changed resolution from "" to "fixed"
* david.schultz changed status from "new" to "closed"

**Comment by icecube-trac** [comment]
Trac update at 20190111T21:01:04: nwhitehorn commented:

> In changeset 168833:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="168833"
> remove icetray memory code. this will be replaced by an external tool using LD_PRELOAD, with one helper function in icetray to change the scope for each module. closes #2075
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190213T14:14:51: 
* olivas commented:

> Milestone December 2017 deleted


* olivas changed milestone from "December 2017" to ""
